code/game/machinery/doors/brigdoors.dm

```diff
@@ -276,71 +287,108 @@
 /obj/machinery/door_timer/attack_hand(mob/user)
 	if(..())
 		return
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/door_timer/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/door_timer/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "brig_timer.tmpl", "Brig Timer", 500, 400)
+		ui = new(user, src, ui_key, "BrigTimer",  name, 500, 450, master_ui, state)
 		ui.open()
-		ui.set_auto_update(TRUE)
-
-/obj/machinery/door_timer/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
-	var/data[0]
-	data["cell_id"] = name
-	data["occupant"] = occupant
-	data["crimes"] = crimes
-	data["brigged_by"] = officer
-	data["time_set"] = seconds_to_clock(time / 10)
-	data["time_left"] = seconds_to_clock(timeleft())
-	data["timing"] = timing
-	data["isAllowed"] = allowed(user)
-	data["prisoner_name"] = prisoner_name
-	data["prisoner_charge"] = prisoner_charge
-	data["prisoner_time"] = prisoner_time
 
+/obj/machinery/door_timer/tgui_static_data(mob/user)
+	var/list/data = list()
+	data["spns"] = list()
```

Comment:
Not a merge-blocking nitpick, but what is this even supposed to mean? Security prisoners near screen?

Reply:
Originally the var was named Suggested_Prisoner_NameS (without the letters capitalized), which exactly describes what the var is - a list of names suggested to the brigging officer to save them time identifying their prisoner.
I ended up changing it to "spns" simply because it was short, and I was tired of TGUI's line length limits forcing me to re-write stuff in a way that took 3-5 more lines than necessary, just because it couldn't handle var names as long as "suggested_prisoner_names" without starting to force every line to be broken up into several lines.